handler/src/main/java/io/netty/handler/pcap/PCapWriter.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.pcap;
+
+import io.netty.buffer.ByteBuf;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+final class PCapWriter implements Closeable {
+    private final SimpleDateFormat sdf = new SimpleDateFormat("SSSSSSSSS");
+    private final OutputStream outputStream;
+
+    /**
+     * This uses {@link OutputStream} for writing Pcap.
+     * Pcap Global Header is also written on construction.
+     *
+     * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
+     */
+    PCapWriter(OutputStream outputStream, ByteBuf byteBuf) throws IOException {
+        this.outputStream = outputStream;
+
+        PcapHeaders.writeGlobalHeader(byteBuf);
+        byteBuf.readBytes(outputStream, byteBuf.readableBytes());
+    }
+
+    /**
+     * Write Packet in Pcap OutputStream.
+     *
+     * @param packetHeaderBuf Packer Header {@link ByteBuf}
+     * @param packet          Packet
+     * @throws IOException If {@link OutputStream#write(byte[])} throws an exception
+     */
+    void writePacket(ByteBuf packetHeaderBuf, ByteBuf packet) throws IOException {
+        long currentTime = System.currentTimeMillis();
+
+        String microsecond = sdf.format(new Date(currentTime));
+        microsecond = microsecond.substring(microsecond.indexOf(".") + 1);
+
+        PcapHeaders.writePacketHeader(
+                packetHeaderBuf,
+                currentTime,
+                Integer.parseInt(microsecond),
```

Comment:
You need two time components here; seconds and the remainder in microseconds. You can do that with arithmetic instead of these shenanigans with date-time formatting, string slicing and number parsing.

To get seconds, you divide by 1000, or use the time unit conversion. And to get microseconds, you first take the modulo of 1000, and then multiply or convert.